src/util/HttpServer/HttpUtils.h

```diff
@@ -76,11 +76,13 @@ static auto createOkResponse(std::string text, const HttpRequest auto& request,
 /// Create a HttpResponse from a stream_generator with status 200 OK.
 static auto createOkResponse(
     ad_utility::stream_generator::stream_generator&& generator,
-    const HttpRequest auto& request, MediaType mimeType) {
+    const HttpRequest auto& request, MediaType mimeType,
+    ad_utility::content_encoding::CompressionMethod method) {
   http::response<ad_utility::httpUtils::httpStreams::streamable_body> response{
       http::status::ok, request.version()};
   response.set(http::field::content_type, toString(mimeType));
   response.keep_alive(request.keep_alive());
+  generator.setContentEncoding(method);
```

Comment:
I currently don't like, that the header field `content-encoding` is implictly set by the `writer` which is automatically called.
Either this has to be commented, or explicitly stated:
`generator.setCompressionMethod(compressionMethod)`
`reponse.setContentEncoding(compressionMethod)`.
But maybe the comment is cleaner.

Reply:
I think the only confusing thing here is the fact that the HttpUtils function does actually call `setCompressionMethod`.
`streamable_body` is simply adapting to `streamable_generator` and setting up the HTTP response accordingly.
If it wasn't setting the reponse headers accordingly then there would be the possibility that someone forgets to set the header when creating a different response and breaks client side, this way there's no way to use `streamable_body` incorrectly.

Reply:
Yes, I agree. Making interfaces that are hard to use wrong is also absolutely my favourite style.
Can you please add a comment here, that the compression method will automagically propagate to the response header's content encoding via the streamable body? (Or whatever is correct:)